### DISCUSSION TITLE
What does community think about passkeys?

### DISCUSSION THREAD

**Discussion by serrq** [Kunzisoft/KeePassDX#1512] [General]
A passwordless vision.

https://fidoalliance.org/fido2/

Possibility or security threat?

**Comment by J-Jamet** [comment]
In study : https://github.com/Kunzisoft/KeePassDX/issues/1421

**Comment by serrq** [comment]
I’m not a developer, I studied philosophy. Please consider my remarks only as those of a non-expert in the field, but only perplexities.

My fear? Actually there are more than one.

The password has no connection to other mathematical values. It is alone and is generated from nothing. The pubkey - privatekey pair has a mathematical correlation instead. True, there is encryption to protect the private key. However, if one day a brute attack breaks the encryption from the public key, you could get the reverse value. And then someone will say: impossible.

In my opinion, there is some mathematical passage, but it is not said that a human being discovers it. There are Ai and they are unpredictable.

My second concern is loyalty.

How do I know that the website I’m asking to create an account does not generate a two-key private keys vault from a single pubkey (both can open the vault) of which I have only one and the other is sent to Intelligence by default?

What if someone plays dirty?

**Reply by J-Jamet** [reply]
> However, if one day a brute attack breaks the encryption from the public key

Did you know that all asymmetric encryption systems are based on a public and private key pair? If from one day to the next we can find a private key from a public key, believe me we will have other problems.

> How do I know that the website I’m asking to create an account does not generate a two-key private keys vault from a single pubkey (both can open the vault) of which I have only one and the other is sent to Intelligence by default?

In passkeys, it is the key provider that provides the public key to the website. I don't know yet the precise technical functioning, because I have to study it, but you can already learn a little bit by reading the articles quoted in the associated issue.

**Reply by serrq** [reply]
> Did you know that all asymmetric encryption systems are based on a public and private key pair? If from one day to the next we can find a private key from a public key, believe me we will have other problems.

How can you be sure that this mathematical passage does not exist? Is there a mathematical countertest that verify it?

However, I agree that there will be other problems if that will happens.

> In passkeys, it is the key provider that provides the public key to the website. I don't know yet the precise technical functioning, because I have to study it, but you can already learn a little bit by reading the articles quoted in the associated issue.

Given my ignorance in cryptography, at the moment the only good questions that comes to mind are: who is the provider of private keys? What role does it play? Should I trust it?»

Of course I’m here to learn and experience, and I like explanations as long as they are within the reach of a non expert. Sorry for my bad English, I still learning it.

**Comment by J-Jamet** [comment]
> How can you be sure that this mathematical passage does not exist? Is there a mathematical countertest that verify it?

Based on prime numbers : https://nrich.maths.org/2200

> Given my ignorance in cryptography, at the moment the only good questions that comes to mind are: who is the provider of private keys? What role does it play? Should I trust it?»


The idea in KeePass will be to store the private keys but other solutions are proposed to the users at first. Google, Microsoft or Apple are in the possibility to be the key providers but as they will be stored on servers of big companies, difficult to trust in this case. That's why we want to store the private keys in KeePass, the problem is the implementation of the WebAuthn protocol which must be studied to be secure and easy to use.

**Reply by serrq** [reply]
Great idea. I think so too. We absolutely must prevent Big Techs from becoming or taking over the private key providers.

My hunch is we won’t have a choice as protocol.

Another "warning" to which I think is: what if in ten years all the most important websites make the traditional password method disappear and leave available only that of passkeys whose providers are the usual known?

How will we defend ourselves?
